snuba/web/rpc/v1/endpoint_get_traces.py

```diff
@@ -0,0 +1,418 @@
+            [self._timer],
+        )
+
+        # Get a dict of trace IDs and timestamps.
+        trace_ids = self._list_trace_ids(request=in_msg)
+        if len(trace_ids) == 0:
+            return GetTracesResponse(meta=response_meta)
+
+        # Get metadata for those traces.
+        traces = self._get_metadata_for_traces(request=in_msg, trace_ids=trace_ids)
+        return GetTracesResponse(
+            traces=traces,
+            page_token=_get_page_token(in_msg, traces),
+            meta=response_meta,
+        )
+
+    def _list_trace_ids(
+        self,
+        request: GetTracesRequest,
+    ) -> dict[str, int]:
+        # Find first span filter.
+        # TODO: support more than one filter.
+        trace_item_filters_expression = trace_item_filters_to_expression(
+            _select_supported_filters(request.filters),
+        )
+        selected_columns: list[SelectedExpression] = [
+            SelectedExpression(
+                name="trace_id",
+                expression=f.cast(
+                    column("trace_id"),
+                    "String",
+                    alias="trace_id",
+                ),
+            ),
+            SelectedExpression(
+                name="_sort_timestamp",
+                expression=f.cast(
+                    column("_sort_timestamp"),
+                    "UInt32",
+                    alias="_sort_timestamp",
+                ),
+            ),
+        ]
+        entity = Entity(
+            key=EntityKey("eap_spans"),
+            schema=get_entity(EntityKey("eap_spans")).get_data_model(),
+            sample=None,
+        )
+        query = Query(
+            from_clause=entity,
+            selected_columns=selected_columns,
+            condition=base_conditions_and(
+                request.meta,
+                trace_item_filters_expression,
+            ),
+            order_by=[
+                OrderBy(
+                    direction=OrderByDirection.DESC,
+                    expression=column("_sort_timestamp"),
+                ),
+            ],
+            limitby=LimitBy(limit=1, columns=[column("trace_id")]),
+            limit=request.limit if request.limit > 0 else _DEFAULT_ROW_LIMIT,
+        )
+
+        treeify_or_and_conditions(query)
+
+        results = run_query(
+            dataset=PluggableDataset(name="eap", all_entities=[]),
+            request=_build_snuba_request(request, query),
+            timer=self._timer,
+        )
+        trace_ids: dict[str, int] = {}
+        for row in results.result.get("data", []):
+            trace_ids[row["trace_id"]] = row["_sort_timestamp"]
+        return trace_ids
+
+    def _get_metadata_for_traces(
+        self,
+        request: GetTracesRequest,
+        trace_ids: dict[str, int],
+    ) -> list[GetTracesResponse.Trace]:
+        trace_item_filters_expression = trace_item_filters_to_expression(
+            request.filters[0].filter if len(request.filters) > 0 else TraceItemFilter()
```

Comment:
did you want to wrap this in `_select_supported_filters` as well? not necessary since the first one already validate it but lets you get rid of this inline ternary